### PR TITLE
Pass exponent as constant instead of device_data

### PR DESCRIPTION
**Pull request by JackCaoG**
This is to implement https://github.com/pytorch/xla/issues/3248.

New HLO for `torch.pow(torch.tensor([2.0, 3.0], device=xm.xla_device()), 5)` looks like
```
IR {
  %0 = f32[] prim::Constant(), location=test_pow_constant@test_operations.py:1770, value=5
  %1 = f32[2]{0} aten::expand(%0), location=test_pow_constant@test_operations.py:1770, size=(2)
  %2 = f32[2]{0} xla::device_data(), location=test_pow_constant@test_operations.py:1770, device=CPU:0
  %3 = f32[2]{0} aten::pow(%2, %1), location=test_pow_constant@test_operations.py:1770, ROOT=0
}
```

Note that `exponent` becomes a `prim::Constant`. This will give XLA compiler more room to optimize and improve the accuracy and speed.

### CODE REVIEW

File: torch_xla/csrc/tensor.h [hunk 1]
```diff
@@ -103,6 +103,10 @@ class XLATensor {
   static ir::Value GetDeviceDataIrValue(const at::Scalar& value,
                                         xla::PrimitiveType type,
                                         const Device& device);
+  // Use with caution, constant will cause more frequent recompilation
+  // compared to the device_data.
+  static ir::Value GetIrValueForConstant(const at::Scalar& value,
```
Comment (yeounoh) [inline torch_xla/csrc/tensor.h:108]: Nit. Should we comment the function def with this info, too, at line 725?

**Review by yeounoh — APPROVED** [review]
LGTM

File: torch_xla/csrc/tensor.h (hunk 1 above)
Comment (JackCaoG) [reply torch_xla/csrc/tensor.h:108]: I think we usually don't repeat the comment in the `cpp` file

**Comment by JackCaoG** [comment]
@yeounoh I am going to merge this pr for now. Let me know if this will impact your experiment.
